### DISCUSSION TITLE
Image | Radxa ROCK 5C / Radxa ROCK 5C Lite

### DISCUSSION THREAD

**Discussion by isarrider** [MichaIng/DietPi#7087] [Image Request]
I just got my Radxa 5C Lites...
(cheap but with A76 cores)
Would be awesome to use them with DietPi...

**Comment by isarrider** [comment]
Update: works with the DietPi_ROCK5A-ARMv8-Trixie Image
I only changed fdtfile=rockchip/rk3588s-rock-5c.dtb from fdtfile=rockchip/rk3588s-rock-5a.dtb in the dietpiEnv.txt

**Comment by isarrider** [comment]
but after trying after apt install linux-image-edge-rockchip-rk3588, the system wont come up again...

**Reply by MichaIng** [reply]
You'd need to install the related DTB package as well:
```sh
apt install linux-image-edge-rockchip-rk3588 linux-dtb-edge-rockchip-rk3588
```
However, I would recommend the `legacy-rk35xx` or `vendor-rk35xx` instead.

**Reply by isarrider** [reply]
Doenst also come up with the DTB...
I want sth later as some of features I wanna test only came in in later kernels...

**Reply by isarrider** [reply]
actually not true - came up but with a different IP now... (other MAC...)

**Reply by MichaIng** [reply]
Random MAC addresses are sadly known for quite a few SBCs. There were fixes for this for some particular SBCs, sometimes particular Ethernet ports only, but not really a generic solution. But if really needed, it can be hardcoded from userland, i.e. udev rule or `/etc/network/interfaces.d` config.

**Reply by isarrider** [reply]
but this only happend when going to Kernel 6.8... Never before, and not since going to 6.8..

**Reply by MichaIng** [reply]
Okay, so a mainline Linux issue in this case.

**Reply by isarrider** [reply]
yap - will see what happens when I try other Rock5Cs....

**Reply by isarrider** [reply]
> You'd need to install the related DTB package as well:
> 
> ```shell
> apt install linux-image-edge-rockchip-rk3588 linux-dtb-edge-rockchip-rk3588
> ```
> 
> However, I would recommend the `legacy-rk35xx` or `vendor-rk35xx` instead.

Why btw - much of rk3588 and so this derivate is going to mainline already...?

**Reply by MichaIng** [reply]
We are not going to mainline Linux. This is just an option. Our RK3588 images are all shipped with `linux-image-legacy-rk35xx`/`linux-dtb-legacy-rk35xx` instead. I aim to migrate to the `vendor` branch (based on Rockchip Linux 6.1 sources), but last time I tested it, I still faced issues, like a udev-worker loop causing constant 5% CPU load.

**Comment by isarrider** [comment]
I dont see any 5% CPU load on my 5B and 5C with 6.8, but ok - thought you had some insights why legacy or vendor is the way... ;)

**Reply by MichaIng** [reply]
A bunch of hardware features, including GPU support, won't work with mainline Linux yet. But it has been a while I tested it. Essential is that onboard WiFi/Ethernet, in case NVMe SSDs and M.2 WiFi modules etc work. I see, the ROCK 5C comes with eMMC slot only, no M.2, so those things should be easy to test.

Would be fine for me to go with mainline Linux for this new SBC, but we cannot migrate other RK3588 SBCs when this breaks too many features for users.

**Comment by isarrider** [comment]
actually I think it is totally ok when we document somewhere how to switch to edge - that way everybody can pick and legacy is standard...?
(In my case for example UART needs a newer one) - shall I do a PR somewhere to document this?

**Reply by MichaIng** [reply]
Since we do not support this SBC yet, not sure where to document it. In general, I am no fans of "supporting" (which is somehow implied when we document how to switch) multiple kernel branches. Our scripts and parts of the setup depend on kernel branch used, e.g. some used sysfs nodes for udev rules and such may change, overlays we apply (rarely but there are cases) do not exist anymore and such. We are too small to properly monitor and have code which correctly handles all kernel branches. Of course we could document it with a "not supported" note, but I prefer to just tell it users directly when being asked, so we can discuss right there how it works, and in case also which parts of DietPi setup/udev rules it broke.

Hence I would just implement it with the one kernel branch we found to be most suitable, and I will migrate other RK3588 SBCs to vendor and/or mainline/edge when the respective issues have been solved/all relevant hardware features are supported, which then implies changes in some scripts and configs here and there.

A particular issue with `edge` kernel is also that it is not bound to LTS, changes rapidly and is mostly untested. So general no great production basis. `current` is mainline Linux as well, but the LTS kernel (Linux 6.6) might not support this SBC (well) yet. Armbian just added that branch for some RK3588 SBCs very recently. We'll see how it works, at least a reason to test it again by times.

**Comment by samolego** [comment]
It's been sometime, @isarrider are you running dietpi on your rock 5c? I just received mine and currently trying to install via conversion script :).

**Comment by isarrider** [comment]
yap, but with the Rock5A image, where I changed in the the dietpiEnv.txt the line from 5a to "fdtfile=rockchip/rk3588s-rock-5c.dtb"

**Reply by samolego** [reply]
Does the hardware work fine? Have you perhaps tested GPU and NPU?

**Reply by isarrider** [reply]
nope, I run it headless as a cheap fast SBC where only the perf and perf/Watt interest me...
So no NPU and GPU tests, sry...

**Reply by samolego** [reply]
Ok, np!

**Reply by samolego** [reply]
It booted successfully, though wifi is not working :disappointed:

**Reply by isarrider** [reply]
I use LAN, so never noticed that

**Comment by Peepo-Juice** [comment]
@MichaIng hello! I just bought the Radxa 5C and it looks like il be using the official Radxa OS for now since DietPi still doesnt support it, but since its been awhile since your last post I was wondering if you could give us an update on that. Is there a blocking issue, or is it still in progress and things are just slow due to being "small"?

Thanks for all the good work you do!

**Comment by isarrider** [comment]
just as a feedback - Rock 5C runs great with the latest Kernel...

<img width="1921" height="2104" alt="image" src="https://github.com/user-attachments/assets/b50910f6-cd7c-4dd1-8f7f-46b517689d81" />
